app/src/main/java/com/alphawallet/app/ui/DappBrowserFragment.java

```diff
@@ -429,44 +433,55 @@ public void onDestroy()
+        final MenuItem scan = toolbar.getMenu().findItem(R.id.action_scan);
+        final MenuItem add = toolbar.getMenu().findItem(R.id.action_add_to_my_dapps);
+
+        if (reload != null) reload.setOnMenuItemClickListener(menuItem -> {
+            reloadPage();
+            return true;
+        });
+        if (share != null) share.setOnMenuItemClickListener(menuItem -> {
+            if (web3.getUrl() != null && currentFragment != null && currentFragment.equals(DAPP_BROWSER)) {
+                if (getContext() != null) viewModel.share(getContext(), web3.getUrl());
+            }
+            else
+            {
+                displayNothingToShare();
+            }
+            return true;
+        });
+        if (scan != null) scan.setOnMenuItemClickListener(menuItem -> {
+            viewModel.startScan(getActivity());
+            return true;
+        });
+        if (add != null) add.setOnMenuItemClickListener(menuItem -> {
+            viewModel.addToMyDapps(getContext(), currentWebpageTitle, urlTv.getText().toString());
+            return true;
+        });
+    }
+
     private void initView(View view) {
         web3 = view.findViewById(R.id.web3view);
         progressBar = view.findViewById(R.id.progressBar);
         urlTv = view.findViewById(R.id.url_tv);
         swipeRefreshLayout = view.findViewById(R.id.swipe_refresh);
         swipeRefreshLayout.setRefreshInterface(this);
         toolbar = view.findViewById(R.id.address_bar);
-        toolbar.inflateMenu(R.menu.menu_bookmarks);
-        toolbar.getMenu().findItem(R.id.action_reload)
-                .setOnMenuItemClickListener(menuItem -> {
-                    reloadPage();
-                    return true;
-                });
-        toolbar.getMenu().findItem(R.id.action_share)
-                .setOnMenuItemClickListener(menuItem -> {
-                    if (web3.getUrl() != null && currentFragment != null && currentFragment.equals(DAPP_BROWSER)) {
-                        if (getContext() != null) viewModel.share(getContext(), web3.getUrl());
-                    }
-                    else
-                    {
-                        displayNothingToShare();
-                    }
-                    return true;
-                });
-        toolbar.getMenu().findItem(R.id.action_scan)
-                .setOnMenuItemClickListener(menuItem -> {
-                    viewModel.startScan(getActivity());
-                    return true;
-                });
-        toolbar.getMenu().findItem(R.id.action_add_to_my_dapps)
-                .setOnMenuItemClickListener(menuItem -> {
-                    viewModel.addToMyDapps(getContext(), currentWebpageTitle, urlTv.getText().toString());
-                    return true;
-                });
+        if (VisibilityFilter.minimiseBrowserURLBar()) toolbar.inflateMenu(R.menu.menu_scan);
+        else toolbar.inflateMenu(R.menu.menu_bookmarks);
```

Comment:
`menu_dapp_browser` instead of `menu_bookmarks` since it's not a menu about bookmarks.